opm/simulators/flow/BlackoilModel_impl.hpp

```diff
@@ -444,18 +446,36 @@ relativeChange() const
             }
             for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++ phaseIdx) {
                 Scalar tmpSat = saturationsNew[phaseIdx] - saturationsOld[phaseIdx];
-                resultDelta += tmpSat*tmpSat;
-                resultDenom += saturationsNew[phaseIdx]*saturationsNew[phaseIdx];
-                assert(std::isfinite(resultDelta));
-                assert(std::isfinite(resultDenom));
+                resultDeltaSaturation += tmpSat*tmpSat * elemVolume;
+                resultDenomSaturation += saturationsNew[phaseIdx]*saturationsNew[phaseIdx] * elemVolume;
+                assert(std::isfinite(resultDeltaSaturation));
+                assert(std::isfinite(resultDenomSaturation));
             }
         }
     }
 
-    resultDelta = gridView.comm().sum(resultDelta);
-    resultDenom = gridView.comm().sum(resultDenom);
-
-    return resultDenom > 0.0 ? resultDelta / resultDenom : 0.0;
+    gridView.comm().sum(r.data(), r.size());
+
+    switch(version) {
+        case RelativeChangeApproach::Pressure:
+            if (resultDenomPressure > 0.0)
+                return std::sqrt(resultDeltaPressure / resultDenomPressure);
+            break;
+        case RelativeChangeApproach::Saturation:
+            if (resultDenomSaturation > 0.0)
+                return std::sqrt(resultDeltaSaturation / resultDenomSaturation);
+            break;
+        case RelativeChangeApproach::PressureSaturation:
+            if (resultDenomPressure > 0.0 && resultDenomSaturation > 0.0) {
+                if (simulator_.timeStepIndex() == 0) {
```

Comment:
Add comment: "Only using pressure for time step zero is a pragmatic choice to avoid excessive cutting for the first step".